### PR TITLE
impl wrap_with_listenable_builder assist

### PR DESCRIPTION
**Pull request by imsamgarg**
# Description

wrap_with_listenable_builder Dart Assist Implementation:

Just like ValueListenableBuilder assist this assist will work in such a way that it only traverses the current node's non-child arguments searching for a single changeNotifier identifier. But the assist will still show even if there's no changeNotifier identifier found. 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor
- [ ] Documentation
- [ ] Other

## Checklist

- [x] I have read the [CONTRIBUTING.md][contributing_link] document.
- [x] I have performed a self-review of my code.
- [ ] I have linked the issue ticket in the description.
- [ ] I have made the necessary changes to the documentation.
- [x] I have checked the formatting with `dart format .`
- [x] I have analyzed my code with `dart analyze .`
- [x] I have run `dart run custom_lint example` to check for any custom linting issues.

[contributing_link]: https://github.com/charlescyt/pyramid_lint/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: lib/src/assists/flutter/wrap_with_listenable_builder.dart [hunk 1]
```diff
@@ -17,7 +18,16 @@ class WrapWithListenableBuilder extends DartAssist {
...
+    // Its not neccessary to have flutter mentioned in the sdk block.
+    final flutterVersion = context.pubspec.environment?['flutter'];
+    final minimumRequiredVersion = Version(3, 10, 0);
+
+    switch (flutterVersion) {
+      case final Version version when version < minimumRequiredVersion:
+      case VersionRange(:final min)
+          when min != null && min < minimumRequiredVersion:
+        return;
+    }
```
Comment (imsamgarg) [inline lib/src/assists/flutter/wrap_with_listenable_builder.dart:30]: I have added the minimum version check but the problem is that there's no guarantee there will be flutter will be mentioned in the SDK block. There's no other way to get the version from the lock file. With the current implementation, we will show the assist if we are not sure about the flutter version. But we can also go the opposite way to only show the assist if we are definite about the flutter version. What do you think?

File: lib/src/assists/flutter/wrap_with_listenable_builder.dart [hunk 2]
```diff
@@ -3,6 +3,7 @@ import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:analyzer/source/source_range.dart';
 import 'package:analyzer_plugin/utilities/range_factory.dart';
 import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:pub_semver/pub_semver.dart';
```
Comment (imsamgarg) [inline lib/src/assists/flutter/wrap_with_listenable_builder.dart:6]: We might also need to include this package in the dependencies.

File: lib/src/assists/flutter/wrap_with_listenable_builder.dart (hunk 1 above)
Comment (charlescyt) [reply lib/src/assists/flutter/wrap_with_listenable_builder.dart:30]: I though about this as well. One way is to maintain a map of Flutter version to Dart version from https://docs.flutter.dev/release/archive?tab=linux.

For example, Flutter 3.10.0 uses Dart 3.0.0. We can check if there is flutter in `dependencies` and the min Dart SDK version in the `environment` instead.

However, this requires more work. We could also go with your approach.

Comment (imsamgarg) [reply lib/src/assists/flutter/wrap_with_listenable_builder.dart:30]: I think we can check for the dart version as you just said which btw is guaranteed to be there in the SDK block.  I'll update the code. What about that pub_semver package. Do we need to include it in the dependencies or just use it without specifying it in the dependencies for now?

Comment (charlescyt) [reply lib/src/assists/flutter/wrap_with_listenable_builder.dart:30]: Yes. Add it to the pubspec if you need it. Otherwise you will get a lint warning.

Comment (imsamgarg) [reply lib/src/assists/flutter/wrap_with_listenable_builder.dart:30]: Yeah sure.

**Pull request merged**